### ISSUE TITLE
Universe inconsistency with QuickChick

### ISSUE THREAD

**Issue by Lysxia** [DeepSpec/InteractionTrees#43]
```coq
Require Import QuickChick.QuickChick. (* current 8.8 branch (idk about the released version) *)
Require Import ITree.ITree.
(* Universe inconsistency *)
```

I think that's somehow caused by the last merge of paco proofs. I will take a closer look later.

**Comment by gmalecha** [comment]
Possibly around `relation`, but I don't know. We don't have anything QuickChick related in the repo, right?

**Comment by Lysxia** [comment]
That's right, it just came up while updating our DeepSpec code.

**Comment by gmalecha** [comment]
`relation` is notoriously bad because it is used everywhere and isn't universe polymorphic.

**Comment by Lysxia** [comment]
Putting `Set Universe Polymorphism.` everywhere seems to do the trick, at least temporarily.

**Comment by Lysxia** [comment]
`Set Universe Polymorphism.` somehow breaks the examples. It seems we need a more fine grained approach.

Changes are in the poly-broken branch https://github.com/DeepSpec/InteractionTrees/tree/poly-broken

**Comment by Lysxia** [comment]
Fixed in https://github.com/DeepSpec/InteractionTrees/commit/f1f9ae40a18e3f13b3882107e4354129345d4d88

For reference, to find the problem, I added `Require Import QuickChick.QuickChick.` to all modules and recompiled itrees to find the breakage, which ended up being only in those containing proofs, and in particular proofs using `dependent destruct`, which seems to be doing strange things with `JMeq`.

Fixing those proofs made the error go away, so we are safe for one more day (at least, hopefully).

More things we could to:

- Separate all the proofs/Prop stuff from the actual computational content. `Fix` is the main remaining piece: there should be a `FixFacts`.

- Remove all occurences of `dependent destruct`.

- Replace `JMeq`/`eq_dep` with universe polymorphic versions. I am not entirely certain they are the real cause, but that's what the error message from `dependent destruct` points to. There are a few uses from the old definition of `eutt` that Gil put aside for convenience, but I believe that on the long term, using these heterogeneous equalities in defining bisimulation (whether strict or up to tau) is actually necessary to avoid relying on proof irrelevance axioms for transitivity.
